client-runtime/src/main/java/com/microsoft/rest/v2/serializer/DateTimeSerializer.java

```diff
@@ -29,17 +29,33 @@
      */
     public static SimpleModule getModule() {
         SimpleModule module = new SimpleModule();
-        module.addSerializer(DateTime.class, new DateTimeSerializer());
+        module.addSerializer(OffsetDateTime.class, new DateTimeSerializer());
         return module;
     }
 
     @Override
-    public void serialize(DateTime value, JsonGenerator jgen, SerializerProvider provider) throws IOException {
+    public void serialize(OffsetDateTime value, JsonGenerator jgen, SerializerProvider provider) throws IOException {
         if (provider.isEnabled(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)) {
-            jgen.writeNumber(value.getMillis());
+            jgen.writeNumber(value.toInstant().toEpochMilli());
         } else {
-            value = value.withZone(DateTimeZone.UTC);
-            jgen.writeString(value.toString(ISODateTimeFormat.dateTime()));
+            jgen.writeString(toString(value));
         }
     }
+
+    /**
+     * Convert the provided OffsetDateTime to its String representation.
+     * @param offsetDateTime The OffsetDateTime to convert.
+     * @return The String representation of the provided offsetDateTime.
+     */
+    public static String toString(OffsetDateTime offsetDateTime) {
+        String result = null;
+        if (offsetDateTime != null) {
+            offsetDateTime = offsetDateTime.withOffsetSameInstant(ZoneOffset.UTC);
+            result = DateTimeFormatter.ISO_INSTANT.format(offsetDateTime);
+            if (result.startsWith("+")) {
```

Comment:
Welp

Reply:
Has anyone out in the wild reported/discussed/rationalized this behavior?

Reply:
The ISO 8601 standard says that outside of 4 digit years any two parties using these serialized strings can determine whether they want to use + in front of their years. Joda and Autorest didn't use + signs, but it looks like the Java OffsetDateTime does use it. I'd rather not, so I just strip the leading + if it exists.